pkg/restclient/errors.go

```diff
@@ -0,0 +1,23 @@
+package restclient
+
+import (
+	"fmt"
+)
+
+// APIUnexpectedStatusCodeError is custom error when expected
+// status code does not match with return status code
+type APIUnexpectedStatusCodeError struct {
+	msg      string
+	expected int
+	actual   int
+	resp     string
+}
+
+func raiseAPIUnexpectedStatusCodeError(expected int, actual int, resp string) error {
```

Comment:
This function isn't actually required. The `msg` is static and isn't actually used when forming the error string. In the one place this is actually being called, you can just create an in-place `UnexpectedStatusError` object.